db/design_doc.go

```diff
@@ -329,7 +329,7 @@ func filterViewResult(input sgbucket.ViewResult, user auth.User, applyChannelFil
 // Is any item of channels found in visibleChannels?
 func channelsIntersect(visibleChannels ch.TimedSet, channels []interface{}) bool {
 	for _, channel := range channels {
-		if visibleChannels.Contains(channel.(string)) || channel == "*" {
+		if visibleChannels.Contains(channel.(string)) {
```

Comment:
@adamcfraser please confirm removing this or condition is okay. I don't understand why the `*` channel was checked in the first place. This fixes `rest/TestUserViewQuery` test which was getting 2 documents but was expecting just one.